solidity/contracts/bridge/Wallets.sol

```diff
@@ -362,19 +372,12 @@ library Wallets {
         bytes20 walletPubKeyHash
     ) internal {
         Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+        // Initialize the moving funds process.
+        wallet.state = WalletState.MovingFunds;
+        /* solhint-disable-next-line not-rely-on-time */
+        wallet.movingFundsRequestedAt = uint32(block.timestamp);
 
-        if (wallet.mainUtxoHash == bytes32(0)) {
-            // If the wallet has no main UTXO, that means its BTC balance
-            // is zero and it should be closed immediately.
-            closeWallet(self, walletPubKeyHash);
```

Comment:
I know we talked about it but I am no longer sure if it is a good move. I would rather transition that wallet to Closing state. See what is happening: we know that the wallet has no funds, yet we ask the wallet to move the funds. The wallet should respond - in a separate transaction - *but I have no funds to move!* and we should close it, but it is also possible, that the wallet [will send the commitment](https://github.com/keep-network/tbtc-v2/pull/208#discussion_r854952434) and then will try to submit a proof of 0-funds moved with no input UTXO and submitting this proof with no main UTXO is not possible:

https://github.com/keep-network/tbtc-v2/blob/5c95ed5c36e440c9aca25ec4f2694f9d8966f84f/solidity/contracts/bridge/MovingFunds.sol#L80-L82

Reply:
According to our discussion https://github.com/keep-network/tbtc-v2/pull/208#discussion_r857396683, I'm keeping this condition in `moveFunds`. In the mentioned discussion, we agreed that `moveFunds` should close all wallets whose balance is below the dust threshold, not just those who have main UTXO unset. Unfortunately, this is not feasible since that needs to pass the wallet main UTXO to resolve the balance and at least one "intention-to-close" transaction (the wallet heartbeat failure) cannot do that easily. However, I think this is not a problem if we rely just on the presence of the main UTXO. That makes the behavior of the system quite consistent:
- `submitMovingFundsCommitment` function reverts if wallet's main UTXO is unset so a zero-balance wallet cannot submit the commitment
- `moveFunds` begins wallet closing in case its main UTXO is unset so a zero-balance wallet will not enter to `MoviungFunds`
- There will be a separate function that will allow closing a wallet with a non-zero but below-dust balance

Reply:
Yeah, I also think it's a consistent approach. The wallet may be non-zero but below the dust threshold at the moment of transitioning to MoveFunds but it could also become non-zero but below the dust threshold after transitioning to MoveFunds, as a result of finalizing all outstanding redemptions. That function we are going to add will serve both cases.

Just one comment though:
> There will be a separate function that will allow closing a wallet with a non-zero but below-dust balance

There is also a possibility that the wallet will end up with zero balance after finalizing all pending redemptions. That new function needs to take this case into account.